packages/zwave-js/src/lib/commandclass/DoorLockLoggingCC.ts

```diff
@@ -0,0 +1,662 @@
+import {
+	CommandClasses,
+	enumValuesToMetadataStates,
+	Maybe,
+	MessageOrCCLogEntry,
+	validatePayload,
+	ValueID,
+	ValueMetadata,
+	ZWaveError,
+	ZWaveErrorCodes,
+} from "@zwave-js/core";
+import { pick } from "@zwave-js/shared";
+import type { Driver } from "../driver/Driver";
+import { MessagePriority } from "../message/Constants";
+import {
+	PhysicalCCAPI,
+	PollValueImplementation,
+	POLL_VALUE,
+	throwUnsupportedProperty,
+} from "./API";
+import {
+	API,
+	CCCommand,
+	CCCommandOptions,
+	ccValue,
+	ccValueMetadata,
+	CommandClass,
+	commandClass,
+	CommandClassDeserializationOptions,
+	expectedCCResponse,
+	gotDeserializationOptions,
+	implementedVersion,
+} from "./CommandClass";
+
+interface DateSegments {
+	year: number;
+	month: number;
+	day: number;
+	hour: number;
+	minute: number;
+	second: number;
+}
+
+function segmentsToDate(segments: DateSegments): Date {
+	return new Date(
+		segments.year,
+		segments.month,
+		segments.day,
+		segments.hour,
+		segments.minute,
+		segments.second,
+	);
+}
+
+// All the supported commands
+export enum DoorLockLoggingCommand {
+	RecordsSupportedGet = 0x01,
+	RecordsSupportedReport = 0x02,
+	RecordGet = 0x03,
+	RecordReport = 0x04,
+}
+
+// @publicAPI
+export enum EventType {
+	LockCode = 0x01,
+	UnlockCode = 0x02,
+	LockButton = 0x03,
+	UnlockButton = 0x04,
+	LockCodeOOSchedule = 0x05,
+	UnlockCodeOOSchedule = 0x06,
```

Comment:
These are already so long, let's write out the "out of":
```suggestion
	LockCodeOutOfSchedule = 0x05,
	UnlockCodeOutOfSchedule = 0x06,
```